blockchain/types/account/externally_owned_account.go

```diff
@@ -67,3 +116,202 @@ func (e *ExternallyOwnedAccount) Equal(a Account) bool {
 }
+
+func (e *ExternallyOwnedAccount) GetStorageRoot() common.ExtHash {
+	return e.storageRoot
+}
+
+func (e *ExternallyOwnedAccount) GetCodeHash() []byte {
+	return e.codeHash
+}
+
+func (e *ExternallyOwnedAccount) GetCodeFormat() params.CodeFormat {
+	return e.codeInfo.GetCodeFormat()
+}
+
+func (e *ExternallyOwnedAccount) GetVmVersion() params.VmVersion {
+	return e.codeInfo.GetVmVersion()
+}
+
+func (e *ExternallyOwnedAccount) SetStorageRoot(h common.ExtHash) {
+	e.storageRoot = h
+}
+
+func (e *ExternallyOwnedAccount) SetCodeHash(h []byte) {
+	e.codeHash = h
+}
+
+func (e *ExternallyOwnedAccount) SetCodeInfo(ci params.CodeInfo) {
+	e.codeInfo = ci
+}
+
+func (e *ExternallyOwnedAccount) toSerializable() *externallyOwnedAccountSerializable {
+	return &externallyOwnedAccountSerializable{
+		CommonSerializable: e.AccountCommon.toSerializable(),
+		StorageRoot:        e.storageRoot.Unextend(),
+		CodeHash:           e.codeHash,
+		CodeInfo:           e.codeInfo,
+	}
+}
+
+func (e *ExternallyOwnedAccount) toSerializableExt() *externallyOwnedAccountSerializableExt {
+	return &externallyOwnedAccountSerializableExt{
+		CommonSerializable: e.AccountCommon.toSerializable(),
+		StorageRoot:        e.storageRoot,
+		CodeHash:           e.codeHash,
+		CodeInfo:           e.codeInfo,
+	}
+}
+
+func (e *ExternallyOwnedAccount) fromSerializable(o *externallyOwnedAccountSerializable) {
+	e.AccountCommon.fromSerializable(o.CommonSerializable)
+	e.storageRoot = o.StorageRoot.ExtendZero()
+	e.codeHash = o.CodeHash
+	e.codeInfo = o.CodeInfo
+}
+
+func (e *ExternallyOwnedAccount) fromSerializableExt(o *externallyOwnedAccountSerializableExt) {
+	e.AccountCommon.fromSerializable(o.CommonSerializable)
+	e.storageRoot = o.StorageRoot
+	e.codeHash = o.CodeHash
+	e.codeInfo = o.CodeInfo
+}
+
+func (e *ExternallyOwnedAccount) EncodeRLP(w io.Writer) error {
+	if e.shouldUseCompactEncoding() {
+		return rlp.Encode(w, e.toSerializable())
+	}
+	return rlp.Encode(w, e.AccountCommon.toSerializable())
+}
+
+func (e *ExternallyOwnedAccount) EncodeRLPExt(w io.Writer) error {
+	var serializable interface{}
+
+	switch {
+	case e.shouldUseCompactEncoding() && e.storageRoot.IsZeroExtended():
+		serializable = e.toSerializable() // Full encoding. [n, b, hR, k, sR, cH, cI]
+	case e.shouldUseCompactEncoding():
+		serializable = e.toSerializableExt() // Full encoding, has ExtHash [n, b, hR, k, sRExt, cH, cI]
+	default:
+		serializable = e.AccountCommon.toSerializable() // Compact encoding. [n, b, hR, k]
+	}
+
+	return rlp.Encode(w, serializable)
+}
+
+func (e *ExternallyOwnedAccount) shouldUseCompactEncoding() bool {
+	zeroHash := common.ExtHash{}
+
+	isEmptyStorageRoot := e.storageRoot == emptyRoot.ExtendZero() || e.storageRoot == zeroHash
+	hasEmptyCodeHash := e.codeHash == nil || bytes.Equal(e.codeHash, emptyCodeHash)
+	hasDefaultCodeInfo := e.codeInfo == params.CodeInfo(0)
+
+	if isEmptyStorageRoot && hasEmptyCodeHash && hasDefaultCodeInfo {
+		return false
+	}
+
+	return true
```

Comment:
This function does the opposite of what its name suggests.
We also need to change the usage of this at the same time. 
```suggestion
	if isEmptyStorageRoot && hasEmptyCodeHash && hasDefaultCodeInfo {
		return true
	}

	return false
```